subprojects/docs/src/docs/release/notes.md

```diff
@@ -69,6 +69,26 @@ ADD RELEASE FEATURES ABOVE
 
 ## New features and usability improvements
 
+### Improved incremental compilation for Java
+
+The Java incremental compiler received substantial improvements in this release:
+
+1. incremental compilation analysis is now stored in the build cache
+2. incremental compilation analysis is faster, uses less memory and disk space
+3. incremental compiler is now aware of constant consumers
```

Comment:
It wasn't immediately clear to me what "incremental compiler is now aware of constant consumers" is about.

How about "incrementation compilation now also works in the presence of constants"?

Reply:
Maybe there is a better way of phrase it but "aware" does not communicate the user value, and "constant consumers" is not entirely clear.

Reply:
mmm, I'm not convinced either. "Presence of constants" may sound like constants in java source files, when the reality is more about constants in upstream dependencies and their use in consumers.

Reply:
What about:
> 3. change of a constant in an upstream dependency doesn't trigger a full recompilation anymore

Reply:
Sounds good to me!